### PR TITLE
unleash-client-cpp: add version 1.3.0, support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **unleash-client-cpp/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/unleash-client-cpp/all/conanfile.py [hunk 1]
```diff
@@ -42,16 +33,15 @@ def _min_cppstd(self):
     def _compilers_min_version(self):
         return {
             "Visual Studio": "15",  # Should we check toolset?
+            "msvc": "190",
```
Comment (valgur) [inline recipes/unleash-client-cpp/all/conanfile.py:36]:
```suggestion
            "msvc": "191",
```
It should probably match the Visual Studio value?

Comment (toge) [reply recipes/unleash-client-cpp/all/conanfile.py]: @valgur 
Oh, it is my falut.
Thanks a lot!

**Review by valgur — APPROVED** [review]

File: recipes/unleash-client-cpp/all/patches/0002-include-sstream.patch
Comment (RubenRBS) [inline recipes/unleash-client-cpp/all/patches/0002-include-sstream.patch:1]: @valgur's patch (and upstream PR) affects more files https://github.com/conan-io/conan-center-index/pull/18739/files#diff-ff2ca12afed223aef2814fbd0c7dacc336e592329f8557d4ea67645a890583ab should we use that one instead?

File: recipes/unleash-client-cpp/all/conanfile.py [hunk 2]
```diff
@@ -42,16 +33,15 @@ def _min_cppstd(self):
     def _compilers_min_version(self):
         return {
             "Visual Studio": "15",  # Should we check toolset?
+            "msvc": "191",
             "gcc": "7",
             "clang": "4.0",
             "apple-clang": "3.8",
             "intel": "17",
```
Comment (RubenRBS) [inline recipes/unleash-client-cpp/all/conanfile.py:40]: `intel` is only v1 - `intel-cc` is for v2, but the versions differ https://docs.conan.io/2/reference/config_files/settings.html (Something that https://github.com/conan-io/conan-center-index/pull/18739/files#diff-e17e1d2d544d2e48d2419318736b0ecbd6ac0e2c6f74b6936759e8358e289112R44 missed)

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! I've seen that @valgur opened a similar PR a few months back (Which fell thru the cracks, sorry!) so let's work on merging both with they corresponding improvements :)

File: recipes/unleash-client-cpp/all/conandata.yml [hunk 3]
```diff
@@ -1,8 +1,5 @@
...
-    url: "https://github.com/aruizs/unleash-client-cpp/archive/refs/tags/v1.1.1.tar.gz"
-    sha256: "2750dc231bf608910d4270ac39d83d46d25b88cc547a9d4d31f7ce4950effa7c"
-patches:
-  "1.1.1":
-    - patch_file: "patches/0001-no-conan-cmake.patch"
-      base_path: "source_subfolder"
+  "1.3.0":
+    url: "https://github.com/aruizs/unleash-client-cpp/archive/refs/tags/v1.3.0.tar.gz"
+    sha256: "fa0b8d6101c6dbd08db23a3d353f386c17e9436a63d658f88c7d0b8619b8d501"
+
```
Comment (RubenRBS) [inline recipes/unleash-client-cpp/all/conandata.yml:5]:
```suggestion
```

**Review by RubenRBS — APPROVED** [review]

**Review by perseoGI — APPROVED** [review]

File: recipes/unleash-client-cpp/config.yml [hunk 4]
```diff
@@ -1,3 +1,4 @@
 versions:
-  "1.1.1":
+  "1.3.0":
     folder: all
+
```
Comment (RubenRBS) [inline recipes/unleash-client-cpp/config.yml:4]:
```suggestion
```

**Review by RubenRBS — APPROVED** [review]

**Review by perseoGI — APPROVED** [review]

**Pull request merged**